plot.go

```diff
@@ -439,49 +439,71 @@ func (p *Plot) NominalY(names ...string) {
 	p.Y.Tick.Marker = ConstantTicks(ticks)
 }
 
-// Save saves the plot to an image file.  The file format is determined
-// by the extension.
+// WriterTo returns an io.WriterTo that will write the plot as
+// the specified image format.
 //
-// Supported extensions are:
+// Supported formats are:
 //
-//  .eps, .jpg, .jpeg, .pdf, .png, .svg, and .tiff.
-func (p *Plot) Save(w, h vg.Length, file string) (err error) {
+//  eps, jpg|jpeg, pdf, png, svg, and tif|tiff.
+func (p *Plot) WriterTo(w, h vg.Length, format string) (io.WriterTo, error) {
```

Comment:
The fluidity is nice, but you lose flexibility over the order of your priorities.

In the case of Save, I decide that if the file can't be openned, there is no point doing the plot work, but I could under other circumstances decide that there is no point getting a writer if the plot work fail. That option is lost.

I'm not sure which is more important here. I'll sleep on it and hear any views you have in the morning.